### ISSUE TITLE
False positive with cppcheck:passedByValue and std::move in constructor

### ISSUE THREAD

**Issue by dfagneray** [SonarOpenCommunity/sonar-cxx#2660]
We have found what we believe is a false positive using Sonarqube and your plugin.

std::move should be more efficient than passing by const reference as suggested

I'm using SonarQube 9.9 with C++ Community 2.1
SonarLint - which IDE/version?
in connected mode with SonarQube or SonarCloud?
How can we reproduce the problem? Give us a self-contained snippet of code (formatted text, no screenshots)
```
#include <string> // Include the <string> header for std::string

class CheckActionAggregator
{
public:
    // Constructor with a string parameter
    explicit CheckActionAggregator(std::string pName)
        : mName(std::move(pName)) // Initialize mName using std::move
    {
        // No additional code here
    }

    // Add any other member functions or variables as needed

private:
    std::string mName;
    // You might want to declare mLogger here if it's a member of CheckActionAggregator
};

int main()
{
    // Create an instance of CheckActionAggregator with a name
    CheckActionAggregator myAggregator("My Check Action");

    // You can now use myAggregator as needed

    return 0;
}
```
The rule will be triggered on the constructor even though there is the std::move which is more efficient than passing by const reference

Can you confirm it's a false positive?
Thanks

**Comment by guwirth** [comment]
Hi @dfagneray,

thanks for your feedback. The cxx plugin does not do static code analysis. The plugin reads only the reports from other tools. In your sample the source of the issue is Cppcheck. Please ask them for help. 

Regards,

**Comment by guwirth** [comment]
Hi @dfagneray,

description for Cppcheck is here:
https://github.com/SonarOpenCommunity/sonar-cxx/wiki/sonar.cxx.cppcheck.reportPaths

But you are running the tool independent of the cxx plugin, means the version depends what’s installed on your machine. 

Regards,
